python/ql/lib/semmle/python/security/dataflow/TarSlipCustomizations.qll

```diff
@@ -0,0 +1,141 @@
+/**
+ * Provides default sources, sinks and sanitizers for detecting
+ * "tar slip"
+ * vulnerabilities, as well as extension points for adding your own.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.Concepts
+private import semmle.python.dataflow.new.BarrierGuards
+private import semmle.python.ApiGraphs
+
+/**
+ * Provides default sources, sinks and sanitizers for detecting
+ * "tar slip"
+ * vulnerabilities, as well as extension points for adding your own.
+ */
+module TarSlip {
+  /**
+   * A data flow source for "tar slip" vulnerabilities.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for "tar slip" vulnerabilities.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A sanitizer for "tar slip" vulnerabilities.
+   */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * A sanitizer guard for "tar slip" vulnerabilities.
+   */
+  abstract class SanitizerGuard extends DataFlow::BarrierGuard { }
```

Comment:
I guess with https://github.com/github/codeql/pull/9590 merged, this will now give a deprecation warning :disappointed: